internal/pkg/api/server/templates/dependencies/dependencies.go

```diff
@@ -68,9 +68,25 @@ func (v *container) CtxCancelFn() context.CancelFunc {
 	return v.ctxCancelFn
 }
 
-func (v *container) Repositories() []model.TemplateRepository {
+func (v *container) Repositories() ([]model.TemplateRepository, error) {
 	// Currently, all projects use the same repositories, in the future it may differ per project.
-	return v.defaultRepositories
+	features, err := v.Features()
+	if err != nil {
+		return nil, err
+	}
+
+	var out []model.TemplateRepository
+	for _, repo := range v.defaultRepositories {
+		if repo.Name == repository.DefaultTemplateRepositoryName && repo.Ref == repository.DefaultTemplateRepositoryRefMain {
+			if features.Has(repository.FeatureTemplateRepositoryBeta) {
+				repo.Ref = repository.DefaultTemplateRepositoryRefBeta
+			} else if features.Has(repository.FeatureTemplateRepositoryDev) {
+				repo.Ref = repository.DefaultTemplateRepositoryRefDev
+			}
+		}
+		out = append(out, repo)
+	}
+	return out, nil
```

Comment:
Načtou se features projektu z tokenu a projdou se `defaultRepositories`. Pokud je v některé z nich repozitář s `name` a `ref` jako defaultní repozitář (což je zatím všude právě jeden který se vrátí z `defaultRepositories`), tak pokud má projekt featuru `templates-repository-dev` vymění se `ref` z `main` za `dev`. A obdobně s featurou `templates-repository-beta`. Všechno je to nadefinovaný jako konstanty v `default.go` níže.